keyboards/handwired/battleship_gamepad/rules.mk

```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no	 # Virtual DIP switch configuration(+1000)
+JOYSTICK_ENABLE = yes    # Enable Joystick HID device
+JOYSTICK_DRIVER = analog # Set joystick reporting to analog
```

Comment:
Even though the [joystick documentation](https://docs.qmk.fm/#/feature_joystick?id=analog-input) does not mention this directly, using `JOYSTICK_DRIVER = analog` also requires enabling the ADC driver, so you also need to add the line specified in the [ADC driver documentation](https://docs.qmk.fm/#/adc_driver?id=usage):
```make
SRC += analog.c
```
Without that line `qmk compile` fails at link time.

Reply:
Thank you.  I was seeing this issue but could not find documentation to fix it.  Maybe we could open an issue to update the joystick documentation.